src/Magpie.Core/EffectCompiler.cpp

```diff
@@ -1373,8 +1374,11 @@ static uint32_t CompilePasses(
 		cbHlsl.append("cbuffer __CB2 : register(b1) { uint __frameCount; };\n\n");
 	}
 
-	if ((flags & EffectCompilerFlags::SaveSources) && !Win32Utils::DirExists(CommonSharedConstants::SOURCES_DIR)) {
-		if (!CreateDirectory(CommonSharedConstants::SOURCES_DIR, nullptr)) {
+	std::wstring sourcesPathName = fmt::format(L"{}{}", CommonSharedConstants::SOURCES_DIR, StrUtils::UTF8ToUTF16(desc.name));
+	std::wstring sourcesPath = sourcesPathName.substr(0, sourcesPathName.find_last_of(L"\\"));
```

Comment:
There is a faster overload of `find_last_of` that takes a wchar_t as input.

Reply:
Fixed.